Support constants in custom storage layout expression (#15944)
ethereum/solidity: Changelog.md

Language Features:

Compiler Features:

> * Custom Storage Layout: the base slot expression can also be specified by constant variables.

Reviewer: ```suggestion
* Custom Storage Layout: The base slot expression can also be specified by constant variables.
```

> * Custom Storage Layout: The base slot expression can also be specified by constant variables.

Reviewer 2: ```suggestion
* Custom Storage Layout: Allow using `constant` state variables in the base slot expression.
```
